src/Shell/Process.php

```diff
@@ -198,7 +208,11 @@ public function __construct(CommandInterface $command, $cwd = null, ProcessOutpu
      */
     public function __destruct()
     {
-        $this->cleanup();
+        try {
+            $this->cleanup();
+        } catch (RuntimeException $e) {
+        } catch (ProcessException $e) {
+        }
```

Comment:
The cleanup method intentionally throws the Process and Runtime exceptions; as this class and it's inheritors are the only ones that can call this method it doesn't make much sense to wrap this method in a try/catch each time it's called.

In the case of the destructor, I agree it makes sense to suppress the exception however, the `wait` function _should not_ suppress the exception cleaning up the process. That failure is something the user should be aware of.

Now, if we're only suppressing the exception in one place the try/catch wrapper is perfectly ok. If we envision needing to toggle the exception suppression further then it might make sense to change the signature of the `cleanup` method to `protected function cleanup($silent = false)` where we suppress exceptions when `$silent` is `true`.

Reply:
in case of wait-method:
the kill method call the cleanup method, and the wait method call the cleanup method again - so we call in case of "kill" the cleanup method twice. also the checkStatus method call the cleanup method and checkStatus is called by isAlive and isAlive is called by wait....

maybe we should reset the "$this->running" variable to false inside the cleanup method, after we call "proc_close"?

Reply:
Yeah, there are a lot of different ways the `cleanup` method ends up getting called.

Honestly, it's been quite a while since I've really used this project, no idea why I added the call to cleanup in `checkStatus`.

Setting `running` to false in the cleanup method works to resolve the potential exception caused when the processed is killed between sequential calls to `cleanup`.